rcl/src/rcl/arguments.c

```diff
@@ -625,6 +624,10 @@ rcl_parse_arguments(
           args_impl->parameter_files[args_impl->num_param_files_args - 1],
           args_impl->num_param_files_args);
         continue;
+      } else if (RCL_RET_ERROR == ret) {
+        // If _rcl_parse_param_file_rule() returned RCL_RET_ERROR then the argument contained the
+        // '__params:=' prefix, but parsing the parameter file failed.
```

Comment:
@jacobperron would it make sense to provide a hint to the user?

Reply:
Good catch. I've copied the error message when when we use the non-deprecated form: f308b49

Also, warning about deprecated usage if there is an error parsing.